bundles/org.openhab.binding.androiddebugbridge/src/main/java/org/openhab/binding/androiddebugbridge/internal/AndroidDebugBridgeDiscoveryService.java

```diff
@@ -0,0 +1,185 @@
+
+import java.io.IOException;
+import java.net.Inet4Address;
+import java.net.InetAddress;
+import java.net.NetworkInterface;
+import java.net.SocketException;
+import java.util.*;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.core.config.discovery.AbstractDiscoveryService;
+import org.openhab.core.config.discovery.DiscoveryResultBuilder;
+import org.openhab.core.config.discovery.DiscoveryService;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingUID;
+import org.osgi.service.cm.Configuration;
+import org.osgi.service.cm.ConfigurationAdmin;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link AndroidDebugBridgeDiscoveryService} discover Android ADB Instances in the network.
+ *
+ * @author Miguel Alvarez - Initial contribution
+ */
+@NonNullByDefault
+@Component(service = DiscoveryService.class, configurationPid = "discovery.androiddebugbridge")
+public class AndroidDebugBridgeDiscoveryService extends AbstractDiscoveryService {
+    static final int TIMEOUT_MS = 60000;
+    private static final long DISCOVERY_RESULT_TTL_SEC = 300;
+    public static final String LOCAL_INTERFACE_IP = "127.0.0.1";
+    public static final int MAX_RETRIES = 2;
+    private final Logger logger = LoggerFactory.getLogger(AndroidDebugBridgeDiscoveryService.class);
+    private boolean discoveryRunning = false;
+    private ConfigurationAdmin admin;
+
+    @Activate
+    public AndroidDebugBridgeDiscoveryService(@Reference ConfigurationAdmin admin) {
+        super(SUPPORTED_THING_TYPES, TIMEOUT_MS, false);
+        this.admin = admin;
+    }
+
+    @Override
+    protected void startScan() {
+        logger.debug("scan started: searching android devices");
+        discoveryRunning = true;
+        Enumeration<NetworkInterface> nets;
+        Dictionary<String, Object> props = getConfig();
+        if (props == null) {
+            return;
+        }
+        var discoveryPort = Integer.parseInt((String) props.get("discoveryPort"));
+        var discoveryReachableMs = Integer.parseInt((String) props.get("discoveryReachableMs"));
+        var discoveryIpRangeMin = Integer.parseInt((String) props.get("discoveryIpRangeMin"));
+        var discoveryIpRangeMax = Integer.parseInt((String) props.get("discoveryIpRangeMax"));
+        try {
+            nets = NetworkInterface.getNetworkInterfaces();
+            for (NetworkInterface netint : Collections.list(nets)) {
+                Enumeration<InetAddress> inetAddresses = netint.getInetAddresses();
+                for (InetAddress inetAddress : Collections.list(inetAddresses)) {
+                    if (!discoveryRunning)
+                        break;
```

Comment:
Please use brackets for every `if` statement.
```suggestion
                    if (!discoveryRunning) {
                        break;
                    }
```